nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -658,3 +664,122 @@ def _fetch_next_batch(self, iterator: Iterator) -> None:
             assert isinstance(dataloader, Sized)  # `_has_len` is True
             self.done = self.fetched >= len(dataloader)
         self.on_fetch_end(batch, start_output)
+
+
+def build_model_cpu(
+    model_provider_func: Callable[[Any, Dict[str, Any]], torch.nn.Module],
+    wrap_with_ddp: bool = True,
+    virtual_pipeline_model_parallel_size: Optional[int] = None,
+    model_type: 'ModelType' = None,  # Default of ModelType.encoder_or_decoder
+    *args: Any,
+    **kwargs: Any,
+) -> List[torch.nn.Module]:
+    """Build the model satisfying pipeline model parallel requirements.
+
+    This function sets `pre_process` and `post_process` to `**kwargs` and pass `*args` and `**kwargs` to
+    `model_provider_func`.
+
```

Comment:
Can you add a note explaining why we need this here?

Reply:
Done